test/aleph/http_test.clj

```diff
@@ -157,6 +164,18 @@
                        :socket-timeout 2000}))]
         (is (= (.replace ^String words "\n" "") (bs/to-string body)))))))
 
+(deftest test-connection-timeout
+  (with-handler basic-handler
+    (is (thrown? TimeoutException
+          @(http/get "http://192.0.2.0" ;; "TEST-NET" in RFC 5737
```

Comment:
Unrelated to my changes, connections to localhost on a closed port seem to come back as established (`conn'` is realized in `http/request`), so we temporarily use a remote host here. I haven't had a chance to look into why this is happening yet.